src/KeyManager.ts

```diff
@@ -331,6 +331,60 @@ class KeyManager {
     })
   }
 
+  // Encrypt data
+  async encryptData(data: Buffer, forPubKey: Buffer): Promise<string> {
+    return new Promise<string>(async (resolve, reject) => {
+      let resolvedIdentity: Object
+      try {
+        resolvedIdentity = await this.getIdentityFromPublicKey(forPubKey)
+      } catch (err) {
+        throw(Error(`Identity could not be resolved for encrypting: ${err}`))
+      }
+      const params = {
+        msg: data,
+        encrypt_for: resolvedIdentity
+      }
+      kbpgp.box(params, (err: Error, result_string: string, result_buffer: Buffer) => {
+        if (err) {
+          reject(err)
+        }
+        resolve(result_string)
+      })
+    })
+  }
+
+  // Encrypt data
+  async decryptData(data: string, withKey?: Buffer): Promise<Buffer> {
+    return new Promise<Buffer>(async (resolve, reject) => {
+      var ring = new kbpgp.keyring.KeyRing;
+      let resolvedIdentity: Object
+      if (withKey) {
+        resolvedIdentity = await this.getIdentityFromPublicKey(withKey)
+      } else if (this.primaryIdentity) {
+        resolvedIdentity = this.primaryIdentity
+      } else {
+        throw(Error('no identity available for signing'))
+      }
+
+      ring.add_key_manager(resolvedIdentity)
+      const params = {
+        armored: data,
+        keyfetch: ring
+      }
+      kbpgp.unbox(params, (err, literals) => {
```

Comment:
If kbpgp takes longer, perhaps we can use the threads.js like we did in js-encryptedfs?

Reply:
I'm not sure web workers would help us with the initial key generation (in polykey initialization) as its really just a one off. I never had an issue with speed on signing and verifying files but we can put it in there :+1: it will probably come in handy later for huge networks as they will be relying on encrypted communication with the users pubkey.

Reply:
A key manager worker has been added in the same way we added one to EFS, it is passed in to the KeyManager class at initialisation via a worker pool (threadsjs)

Reply:
Note this only includes sign/verify and encrypt/decrypt actions for now.